simplify-basedomain-usage/README.md

```diff
@@ -0,0 +1,186 @@
+
+#### Custom Resources triggering changes rather than controllers
+
+Moving configuration from operators to custom resources is always preferable. Starting by the obvious, the cluster configuration is right there explicitly in the `Cluster` Custom resource rather than hidden in controllers configuration. 
+Furthermore, if the value is part of the `Cluster` CR, it's the `Cluster` the one driving and triggering changes on cloud resources, allowing clusters to change or update their values independently of each other.
+This is a better scenario than our current approach having the controller receive this configuration as a parameter, because changing the parameter would trigger changes in cloud resources on all clusters.
+
+#### Reduce coupling
+
+Controllers fetching the `baseDomain` value from the `cluster-values` configmap depend on the `cluster-apps-operator` or `cluster-operator` deployed and creating the configmap.
+Using the value from the annotation instead of depending on the configmap reduces coupling between our controllers.
+
+#### Potential to use different DNS hosted zones on the same Management Cluster
+
+With our current approach, we configure the `baseDomain` at the installation level through the `config` repository, making the `baseDomain` shared for all clusters in a single installation.
+If we move the value to be part of the `Cluster` CR, different clusters could potentially use different hosted zones, even on the same Management Cluster.
+
+#### Migrate workload clusters to different management cluster
+
+One side effect from this is that by decoupling the `baseDomain` from the Management Cluster would also help moving clusters between installations, if that would be desired.
+
+
+### Downsides
+
+#### Migration
+
+For the migration to this approach, we would need to annotate all `Cluster` Custom Resources to have the annotation.
+We also would need to change our operators so that they read this value instead of the parameter they currently receive using the `config` repository.
+
+##### List of controllers that reconcile clusters and use the `baseDomain`
+
+###### aws-resolver-rules-operator
+
+It needs it to construct the k8s api endpoint of the cluster that's reconciling, so that it can create a Route53 Resolver Rule for it.
+
+###### cluster-apps-operator
+
+It needs it to store it in the `cluster-values` configmap that will be consumed by default apps of the cluster that's reconciling.
+
+###### dex-operator
+
+It needs it to configure dex for the cluster that's reconciling. The `baseDomain` passed to the app in the `config` repository is only used in `vintage` management clusters.
+For CAPI clusters or `vintage` workload clusters, the controller loads the `cluster-values` configmap from the management cluster.
+It should also take the value from the annotation instead.
+
+###### dns-operator-*
+
+It needs it to create the hosted zone and DNS records for the cluster that's reconciling.
```

Comment:
There are some stumbling blocks we have to be aware of:

E.g. `mycluser.azuretest.gigantic.io` is set as annotation in the Cluster CR, the `dns-operator` need explicit permissions to create the `NS` records for `mycluster` in the DNS-Zone `azuretest`. For `dns-operator-azure` a dedicated app-credential is needed as the `azuretest` DNS-Zone could exist in another subscription.

If we should support `mycluster.customerproject1.example.com` and `myothercluster.customerproject666.example.com` on azure, we need to know where `customerproject1` and `customerproject666` are hosted and we need explicit permissions to create NS records there (if we are on Azure). For sure, with more far-reaching permissions for the NS record creation or a dedicated list of "supported and granted" DNS zones we can achieve this, but we have to be aware of this and we have to implement this.

Not sure how the `dns-operator` for `aws` or `gcp` is handle such things but i assume, the permission topic might hit every controller somehow.